### ISSUE TITLE
APNG not supported

### ISSUE THREAD

**Issue by javiergutierrezchamorro** [shssoichiro/oxipng#79]
With old OptiPNG I could use -protect acTL,fcTL,fdAT to keep APNG animations. But seems it is not possible with OxiPNG.

Find attached example.

Will you support it?

![animated_png_example_bouncing_beach_ball](https://user-images.githubusercontent.com/4672971/29039372-58f088f0-7bab-11e7-9afc-5de33a4ded54.png)

**Comment by shssoichiro** [comment]
I do plan to implement support for APNG files in Oxipng, however at the moment it is low priority so I can't give a timeframe on when it will be implemented.

**Comment by javiergutierrezchamorro** [comment]
Thanks.
Once implemented, I will include it in FileOptimizer replacing OptiPNG:
http://nikkhokkho.sourceforge.net/static.php?page=FileOptimizer

**Comment by TPS** [comment]
_N.B.: 1st work on this is @ https://github.com/shssoichiro/oxipng/commit/db0131c47e09754777bd642313a0e37a5a910ca3_

**Comment by shssoichiro** [comment]
I just wanted to reassure everyone that I am still working on this issue. I have had less time lately to focus on open source, but I have not abandoned this project.

**Comment by javiergutierrezchamorro** [comment]
Glasd to hear @shssoichiro

**Comment by shssoichiro** [comment]
Unfortunately this is soft-blocked on https://github.com/PistonDevelopers/image/issues/195. The code in oxipng can still be written, but there would be no sanity check on optimized images to make sure oxipng hasn't corrupted them, and no automated tests. I don't feel comfortable delivering this functionality until its correctness can be verified.

**Comment by AlexApps99** [comment]
APNG support was added in image-rs/image#1250, so it's now possible to resume development of this feature.

**Comment by javiergutierrezchamorro** [comment]
Just tried with latest 5.0.0 and reports as not supported yet.

![image](https://user-images.githubusercontent.com/4672971/131451924-b8a5234e-5229-4e46-9035-ee2ffc0dffe4.png)

**Comment by javiergutierrezchamorro** [comment]
Any progress on this?

**Comment by TPS** [comment]
@shssoichiro, @andrews05:  Sorry for tagging y'all directly 🙇🏾‍♂️, but minimal support seems so close.  If y'all could spare a few minutes & let us know whether this is a reasonable idea, we'd appreciate it.:

>  With old OptiPNG I could use -protect acTL,fcTL,fdAT to keep APNG animations. But seems it is not possible with OxiPNG.


 Is it possible, in the interim, to have some way of protecting _just_ these chunks?  IIUC, that would salvage APNGs enough that they could be otherwised optimized by OxiPNG, until full support can be finalized.

**Comment by andrews05** [comment]
@TPS yeah, if #417 gets resolved then we should at least be able to preserve those chunks without simply aborting. I am hoping to solve that issue, so I'll keep this one in mind when looking at it.

**Comment by TPS** [comment]
@andrews05 Thanks very much 🙇🏾‍♂️

**Comment by andrews05** [comment]
@TPS Turns out this isn't quite so simple. The APNG fdAT chunks use the same depth, colour type, palette, interlacing, etc as the main image. If those values are changed at all during optimisation, the fdATs would then be invalid. So it isn't just a matter of preserving those chunks, you would also have to disable all reductions.
If the only possible action then is recompressing the first frame with no reductions, the savings would be very minimal. It may be best to leave this until full support can be implemented.

**Comment by TPS** [comment]
We really appreciate you taking the time to research & explain this, @andrews05 🙇🏾‍♂️

**Comment by TPS** [comment]
I completely agree w/ this issue being closed, but we will need another for tracking **Full aPNG Optimization**.

**Comment by TPS** [comment]
#551 tracks **Full aPNG Optimization**.
